### PR TITLE
Adds freezing to botany

### PR DESCRIPTION
**Pull request by ghost**
You can now freeze botany trays by using a multitool on them.
You will no longer be able to interact with the tray so no harvesting plants or adding water and such.

However, it will stop the plant from using water and nutriment and prevent weed growth, takeover and pest growth.

Trays that are frozen get an updated description, will disable all lights (since you can't use the analyzer since that would take some refactoring to do.) and will turn the plant a slight blue.

You can't freeze trays that are not anchored or that have irrigation systems (as you could add water to them via irrigation.)

### CODE REVIEW

**Comment by myontasm** [comment]
mei joke

**Comment by ArcLumin** [comment]
Botany needs more shit for sure. Too bad there's no way to also increase growth speed but it also increases pest and weed growth

**Comment by ghost** [comment]
Honestly, you could probably add a chem check in the process proc to see if there is X chem or Y feature enabled and have all the procs fire twice as efficiently? It wouldn't be a great deal of effort. Although botany doesn't take too long really. Longest part for me is usually trying to get gud mutations and that's RNG based rather than time or skill.

**Comment by ArcLumin** [comment]
But you get more RNG rolls with faster time

**Comment by ghost** [comment]
Fair point, but wouldn't that be harvest based and do people actually use anything but robust harvest?

**Comment by ArcLumin** [comment]
Let's face it; the issue with botany only shit is that there's loads of plants, but most will only ever make a few varieties due to the RNG needed, and any jobs that need the botany only stuff require lots of it, and a wide variety (I.e. Uranium from plants, fartium, that one cold chem whose name eludes me right now, as well as chef recipes). By making botany much faster, more can be spit out at a time, and there's less "Hurry up and wait" required for botany.

**Comment by KayeArray** [comment]
Perhaps you could just improve the mutagen rng so people spend less time sitting there trying to get a species mutation.

**Comment by ghost** [comment]
Ice chillies is what you're thinking of and yeah, I'd be inclined to agree.
I usually grow grass since it's pretty low maint and use that as biogen fuel for mutagen. Early botany is slow but late botany is pretty fast as you're juggling several plants that you've buffed up with ez mutagen.

**Comment by ArcLumin** [comment]
The idea is a variable speed you can alter depending on how many you're juggling. Focus on one, really fast tray or multiple slower speed trays

**Comment by ghost** [comment]
That'd be an idea to add in a later PR.
Have a selection of chems that swaps the growth mode of the soil.

**Comment by ArcLumin** [comment]
But yeah, I like this tray freezing, it's neat

+1/3rd? Can I even + that much

**Comment by ghost** [comment]
Unforunately not, only GMs count as 1/3.

**Comment by ArcLumin** [comment]
rip. Then
+0.1?
I mean, mentor's gotta mean something
r-right?

**Comment by Badmex** [comment]
Uh okay
+1/3 Why not

File: code/modules/hydroponics/hydroponics.dm [hunk 1]
```diff
@@ -68,9 +89,74 @@
 
 	if(istype(I, /obj/item/weapon/crowbar))
 		if(anchored==2)
-			user << "Unscrew the hoses first!"
+			user << "<span class='warning'>Unscrew the hoses first!</span>"
 			return
 		default_deconstruction_crowbar(I, 1)
+
+	..()
```
Comment (ghost) [inline code/modules/hydroponics/hydroponics.dm:88]: Git is getting confused, this isn't new.

**Comment by ghost** [comment]
+0.1

**Comment by SplergeBucket** [comment]
fuk ye

**Comment by Carbonhell** [comment]
code seems unexpectedly ok
:mememan3:

**Comment by Carbonhell** [comment]
I RETRACT MY STATEMENT
COPYPASTED CODE
HALP

**Comment by ghost** [comment]
I figured out lists, doing a small refactor with overlays.
Also changing it so you can see the plant stats and use analyzers while frozen since I've made it easier to do without snowflaking it.

**Comment by Kokojos** [comment]
Sure +1/3rd gm power.

**Comment by KayeArray** [comment]
Why do we use such an antiquated and archaic numerical rating system to moderate PRs anyway?

**Comment by GenericGoose** [comment]
invent a new one

**Comment by Kokojos** [comment]
Changing my 0.33 vote into a full one. Call loopingh here.

**Comment by ghost** [comment]
" SplergeBucket commented 9 days ago

fuk ye"

I would imagine that's his seal of approval.

**Comment by ArcLumin** [comment]
Or maybe he's a pirate and he meant FUCK YOU

**Comment by Kokojos** [comment]
As good as can be

**Pull request merged**